toolchain/check/context.cpp

```diff
@@ -414,6 +427,20 @@ auto Context::MarkInitializerFor(SemIR::NodeId init_id, SemIR::NodeId target_id)
         }
         return;
       }
+
+      case SemIR::NodeKind::ArrayInit: {
+        // Rewrite the return slot as a reference to our target. We can't just
+        // update the index in `refs_id`, like we do for a Call, because there
+        // will be other references to the return slot for the individual array
+        // element initializers.
+        auto [src_id, refs_id] = init.GetAsArrayInit();
+        semantics_ir().ReplaceNode(
+            semantics_ir().GetNodeBlock(refs_id).back(),
```

Comment:
Why do you check the type of temporary_id in FinalizeTemporary but not here? (I'd have thought this case would be interesting because it's being replaced, so ensuring the right thing is erased)